src/vm/builtins/builtin_runner/ec_op.zig

```diff
@@ -66,26 +75,65 @@ pub const EcOpBuiltinRunner = struct {
         };
     }
 
-    pub fn initSegments(self: *Self, segments: *MemorySegmentManager)  !void {
+    pub fn initSegments(self: *Self, segments: *MemorySegmentManager) !void {
         _ = self;
         _ = segments;
     }
 
     pub fn initialStack(self: *Self, allocator: Allocator) !ArrayList(MaybeRelocatable) {
         _ = self;
         var result = ArrayList(MaybeRelocatable).init(allocator);
-        errdefer result.deinit();        
+        errdefer result.deinit();
         return result;
     }
 
-    pub fn deduceMemoryCell(
-        self: *const Self,
-        address: Relocatable,
-        memory: *Memory,
-    ) ?MaybeRelocatable {
-        _ = memory;
-        _ = address;
-        _ = self;
-        return null;
+    pub fn deduceMemoryCell(self: *Self, allocator: Allocator, address: Relocatable, memory: *Memory) !?MaybeRelocatable {
+        const index = address.offset % self.cells_per_instance;
+        if ((index != OUTPUT_INDICES[0]) and (index != OUTPUT_INDICES[1])) return null;
+
+        const instance = Relocatable.init(address.segment_index, address.offset - index);
+        const x_addr = try instance.addFelt(Felt252.fromInteger(self.n_input_cells));
+
+        if (self.cache.get(address)) |value| {
+            return MaybeRelocatable.fromFelt(value);
+        }
+
+        var input_cells = ArrayList(Felt252).init(allocator);
+
+        // All input cells should be filled, and be integer values.
+        for (0..self.n_input_cells) |i| {
+            if (memory.get(try instance.addFelt(Felt252.fromInteger(i)))) |cell| {
+                const felt = try cell.tryIntoFelt();
+                try input_cells.append(felt);
+            }
+        }
+
+        for (EC_POINTS[0..2]) |pair| {
+            const x = input_cells.items[pair[0]];
+            const y = input_cells.items[pair[1]];
+            var point = EC.ECPoint{ .x = x, .y = y };
```

Comment:
I think we should add an `init` method to `ECPoint` struct in order to avoid this kind of raw initialization wdt?

Reply:
Yes, I will refactor this to use the AffinePoint struct introduced by Nikita